src/api-service/__app__/onefuzzlib/endpoint_authorization.py

```diff
@@ -43,6 +44,35 @@ def is_agent(token_data: UserInfo) -> bool:
     return False
 
 
+def can_modify_config(req: func.HttpRequest, config: InstanceConfig) -> bool:
+    user_info = parse_jwt_token(req)
+    if not isinstance(user_info, UserInfo):
+        return False
+
+    if config.admins is None:
+        return True
+
+    return user_info.object_id in config.admins
+
+
+def check_can_manage_pools(req: func.HttpRequest) -> Optional[Error]:
+    user_info = parse_jwt_token(req)
+    if isinstance(user_info, Error):
+        return user_info
+
+    config = InstanceConfig.fetch()
+    if config.allow_pool_management:
+        return None
+
+    if config.admins is None:
+        return Error(code=ErrorCode.UNAUTHORIZED, errors=["pool modification disabled"])
```

Comment:
If no `admins` are set, then any user should be able to manage pools, right?

So this should be:
```suggestion
        return None
```
We should probably hoist this above the `allow_pool_management` check, too, since that field is only sensible for a config with at least one admin, right?

Reply:
My intent was to soft-disable managing pools.

Reply:
Ah, by "soft-disable" do you mean in a way analogous to admin-overridable GitHub branch protections?

I.e., you want to support this scenario:
> I have a OneFuzz instance with no admin allow-list, so **all** users have admin rights. However, as a footgun avoidance _convention_, I want to disable pool management by default. This convention expects users to temporarily enable pool management, make a pool change, then disable it again.

Reply:
yes.

Reply:
I think it would be helpful to add a comment about that intent.